src/xunit.runner.reporters/VstsReporter.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using Xunit.Abstractions;
+
+namespace Xunit.Runner.Reporters
+{
+    public class VstsReporter : IRunnerReporter
+    {
+        public string Description
+            => "forces VSTS CI mode (normally auto-detected)";
+
+        public bool IsEnvironmentallyEnabled
+            => !string.IsNullOrWhiteSpace(EnvironmentHelper.GetEnvironmentVariable("TF_BUILD")) && 
+               !string.IsNullOrWhiteSpace(EnvironmentHelper.GetEnvironmentVariable("VSTS_ACCESS_TOKEN"));
+
+        public string RunnerSwitch
+            => "vsts";
+
+        public IMessageSink CreateMessageHandler(IRunnerLogger logger)
+        {
+            var accessToken = EnvironmentHelper.GetEnvironmentVariable("VSTS_ACCESS_TOKEN");
+            var collectionUri = EnvironmentHelper.GetEnvironmentVariable("SYSTEM_TEAMFOUNDATIONCOLLECTIONURI");
+            var teamProject = EnvironmentHelper.GetEnvironmentVariable("SYSTEM_TEAMPROJECT");
+
+            //build id is the id associated with the build number, which we will use to associate the test run with
+            var buildId = Convert.ToInt32(EnvironmentHelper.GetEnvironmentVariable("BUILD_BUILDID"));
```

Comment:
Seems like all these environment variables are needed, so shouldn't they all be detected for above?

Reply:
Those are all defined as being present on the build agent by: https://docs.microsoft.com/en-us/vsts/build-release/concepts/definitions/build/variables?tabs=batch

Question is if it's worth checking all of them? The important one is to detect we are running on the agent and that the access token is properly set.

I'm not sure about the name, should we specify `VSTS_ACCESS_TOKEN` or something more specific like `XUNIT_ACCESS_TOKEN`?